cachito/web/models.py

```diff
@@ -680,42 +685,69 @@ class EnvironmentVariable(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String, nullable=False)
     value = db.Column(db.String, nullable=False)
+    kind = db.Column(db.String, nullable=False)
 
-    __table_args__ = (db.UniqueConstraint("name", "value"),)
+    __table_args__ = (db.UniqueConstraint("name", "value", "kind"),)
 
     @classmethod
-    def validate_json(cls, name, value):
+    def validate_json(cls, name, info):
         """
         Validate the input environment variable.
 
-        :param str name: the environment variable name
-        :param str value: the environment variable value
+        :param str name: the name of the environment variable
+        :param dict info: the description of the environment variable. Must include "value" and
+            "kind" attributes
         :raises ValidationError: if the environment variable is invalid
         """
-        if not isinstance(value, str):
+        if not isinstance(name, str):
+            raise ValidationError("The name of environment variables must be a string")
+        if not isinstance(info, dict):
+            raise ValidationError("The info of environment variables must be an object")
+
+        required_keys = {"value", "kind"}
+        missing_keys = required_keys - info.keys()
+        if missing_keys:
+            raise ValidationError(
+                "The following attributes must be set in the info of the environment variables: "
+                f"{', '.join(sorted(missing_keys))}"
```

Comment:
```suggestion
                ", ".join(sorted(missing_keys))
```

Reply:
This doesn't work. It first concatenates the string, then calls join on it.